### DISCUSSION TITLE
How to interact with map using the `leaflet` API?

### DISCUSSION THREAD

**Discussion by xnhp** [vue-leaflet/vue-leaflet#255] [Q&A]
Hi all,

My ultimate goal is to use vue-leaflet together with plugins written for leaflet, in particular [Leaflet.hotline](https://github.com/iosphere/Leaflet.hotline/).

Since I lack the skills and time to wrap this plugin in a `vue-leaflet` component (such as the authors did with `LPolyline` etc), I want to interact with the map in plain-JavaScript-Style and just use the plugin directly.

To this end, I need to be able to interact with the map. I follow [the instructions](https://github.com/vue-leaflet/vue-leaflet/blob/00674e2a20ceca25804dbd6373e0716192a36628/docs/faq/index.md#how-can-i-access-the-leaflet-map-object) to obtain a reference to the map object.

However, when I do

```
    onMapReady() {
      this.mapObject = this.$refs.myMap.leafletObject;
      var latlngs = [
        [45.51, -122.68],
        [37.77, -122.43],
        [34.04, -118.2]
      ];
      var polyline = L.polyline(latlngs, { color: 'red' });
      polyline.addTo(this.mapObject);  // throws error
    },
```

the browser throws

```
Uncaught (in promise) TypeError: max2 is undefined
    intersects Bounds.js:150
    _clipPoints Polyline.js:220
    _update Polyline.js:265
    _reset Path.js:140
    onAdd Path.js:85
    ...
```

`L` is being made available from `import L from 'leaflet'` at the beginning of the script.

Any idea what I'm doing wrong? Thank you very much in advance.

**Comment by xnhp** [comment]
This seems to be mitigated by initialising the map component with

```
<l-map :useGlobalLeaflet="true" ...> ... </l-map>
```

**Comment by mikeu** [comment]
Before the release of version 0.7.0, explicitly setting `:useGlobalLeaflet="true"` was required as you found. That release changed the default behaviour to always work with the global `window.L` instance of Leaflet unless explicitly disabled (with `:useGlobalLeaflet="false"`).
